src/NuGet.Core/NuGet.Commands/RestoreCommand.cs

```diff
@@ -839,7 +860,9 @@ private static PackageDependency GetDependencyVersionRange(LibraryDependency dep
         {
             var range = dependency.LibraryRange.VersionRange;
 
-            if (range == null && dependency.LibraryRange.TypeConstraint == LibraryTypes.ExternalProject)
+            if (range == null
+                && (dependency.LibraryRange.TypeConstraint & LibraryTypeFlag.ExternalProject)
+                    == LibraryTypeFlag.ExternalProject)
```

Comment:
Same here. Perhaps a series of extension methods on whatever type `TypeConstraint` is. It's a little easier to read than bit masking :)